torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -24,6 +26,61 @@ class AllGatherResult(NamedTuple):
     all_gather_input_split_sizes: List[int]
 
 
+lib.define("all_gather_copy_in(SymInt all_gather_input_numel, SymInt world_size, SymInt rank, ScalarType dtype, Device device, SymInt[] inp_split_sizes, Tensor[] all_gather_inputs) -> (Tensor, Tensor)")
+
+@torch.library.impl(lib, "all_gather_copy_in", "Meta")
+def all_gather_copy_in(all_gather_input_numel, world_size, rank, dtype, device, inp_split_sizes, all_gather_inputs):
```

Comment:
Would it be possible to add type annotations to all of the args in these custom op definitions?

Reply:
yup - fixed all type annotations